### ISSUE TITLE
`export { type <ident> }` does not result in `<ident>` being marked as a type

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno#18160]
```ts
// a.ts
export interface A {
  foo: string;
}
```

```ts
// b.ts
import { A } from "./a.ts";
export { type A };
```

#### Actual

```
~/p/g/l/esbuild_deno_loader ❯❯❯ deno run b.ts
error: Uncaught SyntaxError: The requested module './a.ts' does not provide an export named 'A'
import { A } from "./a.ts";
         ^
    at <anonymous> (file:///Users/lucacasonato/projects/github.com/lucacasonato/esbuild_deno_loader/b.ts:1:10)
~/p/g/l/esbuild_deno_loader ❯❯❯ 
```

#### Expected

```
~/p/g/l/esbuild_deno_loader ❯❯❯ deno run b.ts
~/p/g/l/esbuild_deno_loader ❯❯❯ 
```

deno 1.31.1

**Comment by nayeemrmn** [comment]
Fixed by #18795.

**Comment by magurotuna** [comment]
Cool!
